File: 283_moveZeroes_2.java
class Solution {
    public void moveZeroes(int[] nums) {
        int count = 0;
        for (int index = 0; index < nums.length; index++) {
            if (nums[index] == 0) {
                count++;
            } else {
                nums[index-count] = nums[index];
            }
        }
        for (int index = nums.length - count; index < nums.length; index++) {
            nums[index] = 0;
        }
    }
}

File: 283_moveZeroes_2.py
#! /usr/bin/env python3

from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        left = right = 0
        while right < len(nums):
            if nums[right] != 0:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
            right += 1